runtime/near-vm-logic/src/logic.rs

```diff
@@ -781,11 +781,16 @@ impl<'a> VMLogic<'a> {
         let len = value_buf.len() as u64;
         self.gas_counter.pay_per(alt_bn128_g1_multiexp_byte, len)?;
 
-        let discount = (alt_bn128_g1_multiexp_base as u64
-            + alt_bn128_g1_multiexp_byte as u64 * len)
+        let discount = ((alt_bn128_g1_multiexp_base as u64)
+            .checked_add(
+                (alt_bn128_g1_multiexp_byte as u64)
+                    .checked_mul(len)
+                    .ok_or(HostError::IntegerOverflow)?,
+            )
+            .ok_or(HostError::IntegerOverflow)?)
             / alt_bn128_g1_multiexp_sublinear as u64;
         let sublinear_complexity =
-            crate::alt_bn128::alt_bn128_g1_multiexp_sublinear_complexity_estimate(len, discount);
+            crate::alt_bn128::alt_bn128_g1_multiexp_sublinear_complexity_estimate(len, discount)?;
         self.gas_counter.pay_per(alt_bn128_g1_multiexp_sublinear, sublinear_complexity)?;
```

Comment:
The cost formula seems overly complicated. It uses `alt_bn128_g1_multiexp_base` and `alt_bn128_g1_multiexp_byte` (which were already charged before), divides them by `alt_bn128_g1_multiexp_sublinear` and later multiplies it back. In general, all costs should be computed like this: first compute some value that doesn't depend on any costs (e.g. the number of bytes), and then multiply it by a fixed cost value (e.g. cost per byte).